packages/simulator/lib/vehicles/vehicle.js

```diff
@@ -214,10 +223,16 @@ class Vehicle extends EventEmitter {
       metersMoved / 1000,
       (date - this.lastPositionUpdate) / 1000 / 60 / 60,
     ]
-    // https://www.naturvardsverket.se/data-och-statistik/klimat/vaxthusgaser-utslapp-fran-inrikes-transporter/
-    // https://www.trafa.se/globalassets/rapporter/2010-2015/2015/rapport-2015_12-lastbilars-klimateffektivitet-och-utslapp.pdf
-    const co2 = (this.weight + this.cargoWeight()) * km * this.co2PerKmKg
-    this.co2 += co2
+
+    const co2 = this.updateCarbonDioxide(km)
+
+    // TODO: Find which kommun the vehicle is moving in now and add the co2 for this position change to that kommun
```

Comment:
In order to summarize co2 emissions withing a kommun, we need to figure out where this current position change is taking place. I.e. inside which kommun it is and then add the co2 emission diff to that kommun.

It is not enough to simply summarize the co2 for each vehicle that is in a kommun at any given time because the vehicles can - and will - cross the kommun borders.

This issue stems from the fact that vehicles are generated on the region level and not kommun level.

Reply:
My suggestion is to emit a co2 event and group them on kommun in the same way as we do with the rest of the statistics?